# HG changeset patch
# Parent da33c86d96c28fe921cd09438a69531cad013277
Fix 32 bit integer overflow in jsonrpc_float_range_to_uint16

diff --git a/core/jsonrpc.c b/core/jsonrpc.c
--- a/core/jsonrpc.c
+++ b/core/jsonrpc.c
@@ -218,7 +218,7 @@
             fracpart += (s[i] - '0') * multiplier;
         }
     }
-    return ((intpart + fracpart) * UINT16_MAX) / range;
+    return ((int64_t)intpart + (int64_t)fracpart) * UINT16_MAX / (int64_t)range;
 }
 
 void
